Backend refactoring

ho-stanley opened the pull request:
All important endpoint works now.
Readme API documentation is not yet up to date but I thought it would still be good to deploy the working code as soon as possible and have it available for the frontend team.

ho-stanley committed:
Refactor backend

ho-stanley committed:
Fix request user id

Returns a correct JSON message if user is not found.

Co-authored-by: Tuliany Grande <7564014+Tuliany@users.noreply.github.com>
Co-authored-by: Jonas Lund <70700290+JonasLundChas@users.noreply.github.com>

Tuliany committed:
added a readme-file for backend

ho-stanley committed:
Add value range for rating

Value range is 1-10.
Ratings also fetch made by user info.

Closes #15

Co-authored-by: Tuliany Grande <7564014+Tuliany@users.noreply.github.com>
Co-authored-by: Jonas Lund <70700290+JonasLundChas@users.noreply.github.com>
Co-authored-by: Cao <55875337+Caoabc123@users.noreply.github.com>

ho-stanley committed:
Add admin role in user model

Closes #13

Co-authored-by: Tuliany Grande <7564014+Tuliany@users.noreply.github.com>
Co-authored-by: Jonas Lund <70700290+JonasLundChas@users.noreply.github.com>
Co-authored-by: Cao <55875337+Caoabc123@users.noreply.github.com>

ho-stanley committed:
Remove accesstoken from user model

Closes #14

Co-authored-by: Tuliany Grande <7564014+Tuliany@users.noreply.github.com>
Co-authored-by: Jonas Lund <70700290+JonasLundChas@users.noreply.github.com>
Co-authored-by: Cao <55875337+Caoabc123@users.noreply.github.com>

ho-stanley committed:
Update address field in user model

ho-stanley committed:
Update backend readme

Add public routes API documentation.

ho-stanley committed:
Merge branch 'BackendReadME' into backend-refactoring

Add backend readme.

ho-stanley committed:
Merge branch 'rating-value-range' into backend-refactoring

Add value range for rating

Closes #15

ho-stanley committed:
Merge branch 'update-user-model' into backend-refactoring

Update user model

Closes #13 #14

ho-stanley committed:
Refactor signup and get all ratings route

You can pass description and address fields during signup. Get all
ratings from a user response have more detailed data.

ho-stanley committed:
Update public routes

Moved signup and signin endpoint to public route.
Able to update user profile and give/update rating.

Closes #19

Co-authored-by: Tuliany Grande <7564014+Tuliany@users.noreply.github.com>
Co-authored-by: Jonas Lund <70700290+JonasLundChas@users.noreply.github.com>
Co-authored-by: Cao <55875337+Caoabc123@users.noreply.github.com>

ho-stanley committed:
Merge branch 'public-routes' into backend-refactoring

Update public routes.
Able to update user profile and give/update rating.

Co-authored-by: Tuliany Grande <7564014+Tuliany@users.noreply.github.com>
Co-authored-by: Jonas Lund <70700290+JonasLundChas@users.noreply.github.com>
Co-authored-by: Cao <55875337+Caoabc123@users.noreply.github.com>

axelra82 commented on backend/README.md:
```diff
@@ -0,0 +1,202 @@
+
+### Signup
+
+Method: `POST`<br>
```
<br>? 🤔

ho-stanley replied:
It's to add a new line directly below. I could have used two spaces but thought it was not clear when viewing the file.

axelra82 commented on backend/README.md:
````diff
@@ -0,0 +1,202 @@
+```js
+{
+	"status": 200,
+	"token": "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJlbWFpbCI6Im1AbS5jb20iLCJpYXQiOjE2MjE5MzQ1NTI4NjcsImV4cCI6MTYyMTkzODE1Mjg2N30.QnAfjfydKa4-Z2aLRuNFw8co6G9Oev8TEu3Aiyt3QVE",
````
Think about not showing sequences or other possible data that could in some form help potential hackers.

ho-stanley replied:
Noted and changed.

axelra82 commented on backend/models/address.js:
```diff
@@ -0,0 +1,22 @@
+	},
+	zip: {
+		type: Number,
+		max: 20,
```
Look in to `max` vs `maxlength` and `String` vs `Number` for `zip`

ho-stanley replied:
Noted and fixed.

ho-stanley committed:
Remove max number limit on zip field in address

ho-stanley committed:
Add hairdresser filtering

Able to search for hairdressers and filter them based on request
queries. These fields are street, city and zip.

axelra82 commented on backend/auth/auth.js:
```diff
@@ -130,40 +71,135 @@ exports.userById = async (req, res, next) => {
+
+	if (time < payload.exp) {
+		const updateUserId = req.params.id;
+		const id = req.user._id.toString();
```
`_id ` is unknown to frontend. User model does not contain `_id`, hence user can't set rating. Currently email (available in JWT) is unique identifier used in frontend.

Frontend and backend need to be unambiguous on how to identify users.

axelra82 reviewed (changes requested):
~~strike comment~~

ho-stanley committed:
Update address schema

Zip field have correct max value now.

ho-stanley committed:
Update readme

All public API routes are documented.
